spring-tx/src/main/java/org/springframework/dao/support/PersistenceExceptionTranslationInterceptor.java

```diff
@@ -167,8 +172,20 @@ protected PersistenceExceptionTranslator detectPersistenceExceptionTranslators(L
 		// Find all translators, being careful not to activate FactoryBeans.
 		Map<String, PersistenceExceptionTranslator> pets = BeanFactoryUtils.beansOfTypeIncludingAncestors(
 				beanFactory, PersistenceExceptionTranslator.class, false, false);
+
+		Comparator<Object> comparatorToUse = null;
+		if (beanFactory instanceof DefaultListableBeanFactory) {
+			comparatorToUse = ((DefaultListableBeanFactory) beanFactory).getDependencyComparator();
+		}
+		if (comparatorToUse == null) {
+			comparatorToUse = OrderComparator.INSTANCE;
+		}
```

Comment:
I think the above code can be more compact and elegant as
```
Comparator<Object> comparatorToUse = OrderComparator.INSTANCE;
if (beanFactory instanceof DefaultListableBeanFactory) {
	comparatorToUse = ((DefaultListableBeanFactory) beanFactory).getDependencyComparator();
}
```
or simply
```
Comparator<Object> comparatorToUse;
if (beanFactory instanceof DefaultListableBeanFactory) {
	comparatorToUse = ((DefaultListableBeanFactory) beanFactory).getDependencyComparator();
}
else  {
	comparatorToUse = OrderComparator.INSTANCE;
}
```

Reply:
`DefaultListableBeanFactory#getDependencyComparator()` is  `@Nullable`, so it might return `null`.